rest/kafka/consumers.go

```diff
@@ -44,21 +44,36 @@ func startKafkaReader(r *kafka.Reader) {
 			if err != nil {
 				log.Println("Unable marshal payload data", p, err)
 			} else {
-				newMessage := connectionhub.Message{
-					Destinations: connectionhub.MessageDestinations{
-						Users:         p.Data.Users,
-						Roles:         p.Data.Roles,
-						Organizations: p.Data.Organizations,
-					},
-					Broadcast: p.Data.Broadcast,
-					Data:      data,
-				}
-				if p.Data.Broadcast {
-					logrus.Infoln("Emitting new broadcast message from kafka reader: ", string(newMessage.Data))
-					connectionhub.ConnectionHub.Broadcast <- newMessage
+				err := p.DataContentType.IsValid()
+				if err != nil {
+					log.Println("Kafka message payload needs to be JSON formatted", err)
 				} else {
-					logrus.Infoln("Emitting new message from kafka reader: ", string(newMessage.Data))
-					connectionhub.ConnectionHub.Emit <- newMessage
+					err := p.SpecVersion.IsValid()
```

Comment:
I am wondering if we can just exit early from the function rather than keep nesting the code deeper and deeper. Its a bit hard to read.